runhouse/resources/hardware/launcher_utils.py

```diff
@@ -14,6 +16,45 @@
 logger = get_logger(__name__)
 
 
+class LogProcessor:
+    """Dedicated logger for handling streamed cluster logs"""
+
+    def __init__(self):
+        self.cluster_logger = self._init_cluster_logger()
+
+    def _init_cluster_logger(self):
+        cluster_logger = logging.getLogger("cluster_logs")
+
+        cluster_logger.setLevel(logging.DEBUG)
+
+        # Add a handler with a simple formatter if not already set
+        if not cluster_logger.handlers:
+            handler = logging.StreamHandler(sys.stdout)
+            handler.setFormatter(logging.Formatter("%(message)s"))
+            cluster_logger.addHandler(handler)
+
+        # Prevent propagation to root_logger
+        cluster_logger.propagate = False
+
+        return cluster_logger
+
+    def log_event(self, log_message: str, cluster_name: str):
+        """Log the event at the appropriate level using the cluster logger."""
+        # Note: we need to remove logger prefix from the local logger to just include the log in Den
+        log_message = f"[{cluster_name}] {log_message}"
+
+        if "INFO |" in log_message:
```

Comment:
Alternatively, we could potentially use event names for each `log_{type}` - would move some of the logic up to Den/Launcher

Reply:
yea good call that would be a bit cleaner